src/webgpu/api/validation/createView.spec.ts

```diff
@@ -1,311 +1,318 @@
+    const texture = t.device.createTexture({
+      format: 'rgba8unorm',
+      size: [4, 4, 18],
+      usage: GPUTextureUsage.SAMPLED,
+    });
 
-    const descriptor = { format, dimension, arrayLayerCount, mipLevelCount };
+    let success = arrayLayerCount > 0;
+    if (dimension === '2d') {
+      success &&= arrayLayerCount === 1;
+    } else if (dimension === 'cube') {
+      success &&= arrayLayerCount === 6;
+    } else if (dimension === 'cube-array') {
+      success &&= arrayLayerCount % 6 === 0;
+    }
 
     t.expectValidationError(() => {
-      texture.createView(descriptor);
-    }, !_success);
+      texture.createView({ dimension, arrayLayerCount });
+    }, !success);
   });
 
-g.test('Using_defaults_validates_the_same_as_setting_values_for_only_1_array_layer')
-  .paramsSimple([
-    { _success: true },
-    { format: 'rgba8unorm', _success: true },
-    { format: 'r8unorm', _success: false },
-    { dimension: '2d-array', _success: true },
-    { dimension: '2d', _success: true },
-    { arrayLayerCount: 0, _success: true },
-    { arrayLayerCount: 1, _success: true },
-    { arrayLayerCount: 2, _success: false },
-    { mipLevelCount: MIP_LEVEL_COUNT, _success: true },
-    { mipLevelCount: 1, _success: true },
-  ] as const)
-  .fn(async t => {
-    const { format, dimension, arrayLayerCount, mipLevelCount, _success } = t.params;
+g.test('array_layers')
+  .desc(
+    `Views must have at least one layer, and must be within the layers of the base texture.
 
-    const texture = t.createTexture({ arrayLayerCount: 1 });
+  - arrayLayerCount=0 at various baseArrayLayer values
+  - Cases where baseArrayLayer+arrayLayerCount goes past the end of the texture
+  - Cases with baseArrayLayer or arrayLayerCount undefined (compares against reference defaulting impl)
+  `
+  )
+  .paramsSubcasesOnly(u =>
+    u
+      .combineWithParams([
+        { textureLayers: 1, textureLevels: 1 },
+        { textureLayers: kLayers, textureLevels: kLevels },
+      ])
+      .combine('view', [
+        {},
+        { baseArrayLayer: 0 },
+        { arrayLayerCount: 1 },
+        { baseArrayLayer: 0, arrayLayerCount: 1 },
+        { baseArrayLayer: 1, arrayLayerCount: 1 },
+        { baseArrayLayer: kLayers - 2, arrayLayerCount: 1 },
+        { baseArrayLayer: kLayers - 1, arrayLayerCount: 1 },
+        { baseArrayLayer: 0, arrayLayerCount: 2 },
+        { baseArrayLayer: 1, arrayLayerCount: 2 },
+        { baseArrayLayer: kLayers - 2, arrayLayerCount: 2 },
+        // arrayLayerCount == undefined means to use all remaining layers
+        { arrayLayerCount: undefined, baseArrayLayer: 0 },
+        { arrayLayerCount: undefined, baseArrayLayer: 1 },
+        { arrayLayerCount: undefined, baseArrayLayer: kLayers - 1 },
+        { arrayLayerCount: undefined, baseArrayLayer: kLayers },
+        // arrayLayerCount == 0 means zero array layers, which is never valid
+        { arrayLayerCount: 0, baseArrayLayer: 0 },
```

Comment:
Looks like we need to add one condition for `success`: let success = reified.arrayLayerCount > 0 && ...? I guess this test can pass because viewDimension is always `2d` so reified.arrayLayerCount must be 1.